src/Libraries/RevitNodesUI/RevitDropDown.cs

```diff
@@ -745,6 +788,10 @@ public class StructuralFramingTypes : AllElementsInBuiltInCategory
     {
         public StructuralFramingTypes()
             : base(BuiltInCategory.OST_StructuralFraming, "Framing Types", Properties.Resources.DropDownNoFramingType){}
+
+        [JsonConstructor]
+        public StructuralFramingTypes(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) 
+            : base(BuiltInCategory.OST_StructuralFraming, "Framing Types", Properties.Resources.DropDownNoFramingType, inPorts, outPorts) { }
```

Comment:
same for all these - can we pull the strings out into constants in the class?

Reply:
I believe in most if not call cases at least for all the dropdown UI nodes the string being passed in always a hardcoded string referring to the output type so yes they can be pulled out into constants